src/impl/epoll/external_commit_helper.cpp

```diff
@@ -198,14 +253,22 @@ void ExternalCommitHelper::listen()
         if (ev.data.u32 == (uint32_t)m_shutdown_read_fd) {
             return;
         }
-        assert(ev.data.u32 == (uint32_t)m_notify_fd);
 
-        m_parent.on_change();
+        {
+            std::lock_guard<std::mutex> lock(m_mutex);
+            for (auto it : m_helper_list) {
+                if (ev.data.u32 == (uint32_t)it->m_notify_fd) {
+                    it->m_parent.on_change();
```

Comment:
It seems in the notifier thread, create/free external_commit_helper should never happen. Added an assert to detect that.